gluonnlp/data/dataloader.py

```diff
@@ -22,7 +22,10 @@
 import sys
 import multiprocessing
 import multiprocessing.queues
-from mxnet.gluon.data.dataloader import Queue, SimpleQueue, DataLoader, _MultiWorkerIter
+import threading
+from mxnet import context
+from mxnet.gluon.data.dataloader import Queue, SimpleQueue, DataLoader, \
```

Comment:
I suggest to put them in the iterator otherwise gluonnlp will instantly depend on the latest mxnet master

Reply:
Yeah, you are right

Reply:
Or we can let DataLoader takes function worker_loop as an argument in init function. This can makes the code cleaner.

Reply:
@zhreshold gluon-nlp already depends on mxnet master, so it's no problem that this change also requires mxnet master. As far as I understand, your change in mxnet will be part of the 1.3 release?

After the coming release, it would be great if we can reduce the duplication of DataLoader code in gluonnlp. Does @szhengac's suggestion sound good to you?

Reply:
I would agree to add an more customizable option to data loader iter, in terms of every aspect. 
@szhengac Yes you are good to go with the mxnet, let me know if you have concerns.